internal/providers/terraform/azure/function_app.go

```diff
@@ -0,0 +1,188 @@
+package azure
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/infracost/infracost/internal/schema"
+	"github.com/shopspring/decimal"
+	"github.com/tidwall/gjson"
+)
+
+func GetAzureRMAppFunctionRegistryItem() *schema.RegistryItem {
+	return &schema.RegistryItem{
+		Name:  "azurerm_function_app",
+		RFunc: NewAzureRMAppFunction,
+		ReferenceAttributes: []string{
+			"app_service_plan_id",
+		},
+	}
+}
+
+func NewAzureRMAppFunction(d *schema.ResourceData, u *schema.UsageData) *schema.Resource {
+	meterNameCPU := "vCPU Duration"
+	meterNameMemory := "Memory Duration"
+	meterNameExecTime := "Execution Time"
+	meterNameExecutions := "Total Executions"
+	memorySize := 1.0
+	executionTime := 1.1
+	executions := 1.1
+	kind := "kind"
+	skuTier := "tier"
+	skuSize := "EP"
+	skuMemory := 3.0
+	skuCPU := 1
+	instances := 1.0
```

Comment:
These should default to nil so if the user does not define them in the usage file, then we show "cost depends on usage" in the output, not 0 cost.